### PR TITLE
Add campaign leaderboard routes

### PR DESCRIPTION
**Pull request by hectorgomezv**
## Changes
- Adds `GET /v1/locking/campaigns/:campaignId/leaderboard` endpoint mapping.

### CODE REVIEW

File: src/domain/interfaces/locking-api.interface.ts [hunk 1]
```diff
@@ -20,6 +21,12 @@ export interface ILockingApi {
     offset?: number;
   }): Promise<Page<Rank>>;
 
+  getLeaderboardV2(args: {
```
Comment (hectorgomezv) [inline src/domain/interfaces/locking-api.interface.ts:24]: Note that the Safe Locking Service implementation changed recently, and the leaderboard by campaign endpoint is not under the `v2` prefix anymore. These changes can be seen here (approval is pending): https://github.com/safe-global/safe-locking-service/pull/100/files#diff-d790dada13951d227666cb46aa0ccc89033037f7a7a5e1ed86e9bbfbbe0ef65b

A follow-up PR will be created to modify the datasource endpoints and rename functions/variables accordingly.

File: src/domain/locking/locking.repository.interface.ts [hunk 2]
```diff
@@ -20,6 +21,12 @@ export interface ILockingRepository {
     offset?: number;
   }): Promise<Page<Rank>>;
 
+  getLeaderBoardByCampaignId(args: {
```
Comment (iamacook) [inline src/domain/locking/locking.repository.interface.ts:24]: Nit: "leaderboard" is one word:

```suggestion
  getLeaderboardByCampaignId(args: {
```

File: src/domain/locking/locking.repository.ts [hunk 3]
```diff
@@ -46,6 +50,15 @@ export class LockingRepository implements ILockingRepository {
     return RankPageSchema.parse(page);
   }
 
+  async getLeaderBoardByCampaignId(args: {
```
Comment (iamacook) [inline src/domain/locking/locking.repository.ts:53]: As before regarding "leaderboard":
```suggestion
  async getLeaderboardByCampaignId(args: {
```

File: src/routes/locking/entities/holder.entity.ts [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import { Holder as DomainHolder } from '@/domain/locking/entities/holder.entity';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class Holder implements DomainHolder {
```
Comment (iamacook) [inline src/routes/locking/entities/holder.entity.ts:4]: What do you think about calling this `CampaignRank`? This is similar to the existing `Rank` of locking events (and we could rename that later to `LockRank` or something similar). If you prefer "holder", we could aim for `CampaignHolder` and then follow the same adjustment as above for `LockHolder`.

File: src/routes/locking/entities/holder.page.entity.ts [hunk 5]
```diff
@@ -0,0 +1,8 @@
+import { Page } from '@/routes/common/entities/page.entity';
+import { Holder } from '@/routes/locking/entities/holder.entity';
+import { ApiProperty } from '@nestjs/swagger';
+
+export class HolderPage extends Page<Holder> {
```
Comment (iamacook) [inline src/routes/locking/entities/holder.page.entity.ts:5]: We'd need to rename this if you agree with the above as well.

File: src/routes/locking/locking.controller.ts [hunk 6]
```diff
@@ -42,6 +43,25 @@ export class LockingController {
...
   }
 
+  @ApiOkResponse({ type: HolderPage })
+  @ApiQuery({
+    name: 'cursor',
+    required: false,
+    type: String,
+  })
+  @Get('/campaigns/:campaignId/leaderboard')
+  async getLeaderBoardByCampaignId(
```
Comment (iamacook) [inline src/routes/locking/locking.controller.ts:53]: As before regarding "leaderboard":

```suggestion
  async getLeaderboardByCampaignId(
```

File: src/routes/locking/locking.service.ts [hunk 7]
```diff
@@ -68,6 +69,31 @@ export class LockingService {
     };
   }
 
+  async getLeaderBoardByCampaignId(args: {
```
Comment (iamacook) [inline src/routes/locking/locking.service.ts:72]: As before regarding "leaderboard":
```suggestion
  async getLeaderboardByCampaignId(args: {
```

File: src/domain/locking/locking.repository.interface.ts (hunk 2 above)
Comment (hectorgomezv) [reply src/domain/locking/locking.repository.interface.ts]: I'm resolving this one as this got renamed to `getCampaignLeaderboard`. Lmk if you have other suggestion of course!

File: src/routes/locking/locking.controller.ts (hunk 6 above)
Comment (hectorgomezv) [reply src/routes/locking/locking.controller.ts]: I'm resolving this one as this got renamed to `getCampaignLeaderboard`. Lmk if you have other suggestion of course!

File: src/routes/locking/locking.service.ts (hunk 7 above)
Comment (hectorgomezv) [reply src/routes/locking/locking.service.ts]: I'm resolving this one as this got renamed to `getCampaignLeaderboard`. Lmk if you have other suggestion of course!

File: src/routes/locking/entities/holder.entity.ts (hunk 4 above)
Comment (hectorgomezv) [reply src/routes/locking/entities/holder.entity.ts:4]: Totally agree, will rename it to `CampaignRank`, thanks!

Comment (hectorgomezv) [reply src/routes/locking/entities/holder.entity.ts]: Done in [d103261](https://github.com/safe-global/safe-client-gateway/pull/1557/commits/d103261dc2f90845eadbf8913523842a099ce968)

File: src/routes/locking/entities/holder.page.entity.ts (hunk 5 above)
Comment (hectorgomezv) [reply src/routes/locking/entities/holder.page.entity.ts]: Changed in [d103261](https://github.com/safe-global/safe-client-gateway/pull/1557/commits/d103261dc2f90845eadbf8913523842a099ce968)

**Pull request merged**